src/TaskRunner/Commands/ComponentCheckCommands.php

```diff
@@ -172,6 +175,39 @@ protected function prepareSkips(): void
         }
     }
 
+    /**
+     * Validate project configuration.
+     */
+    protected function componentConfiguration()
+    {
+        // Make sure forbidden files does not exist.
+        // (because of deprecation or another reason)
+        // Get forbidden/deprecated files from configuration.
+        $files = $this->getConfig()->get('toolkit.forbidden_files');
+        // Detect forbidden files in project.
+        foreach ($files as $file) {
+            // Check if the file must be forbidden under the condition.
+            if (!empty($file['condition_callback'])) {
+                list($class, $method) = explode('::', $file['condition_callback']);
+                if (class_exists($class) && method_exists($class, $method)) {
+                    // Invoke the callback method and if the condition is not met then don't forbid the file.
+                    if ((new $class())->$method() === false) {
+                        continue;
+                    }
+                }
+            }
+            if (file_exists($this->getWorkingDir() . '/' . $file['name'])) {
+                $this->composerFailed = true;
+                $this->io->error($file['error']);
+            }
+        }
+
+        if (!$this->composerFailed) {
```

Comment:
If this is a new section in the component-check, I will suggest to have its own flag like `$this->configurationFailed` to avoid misunderstanding.

Reply:
Hi @joaocsilva this has been fixed.